app/models/test.rb

```diff
@@ -4,7 +4,16 @@ class Test < ApplicationRecord
   belongs_to :category
   belongs_to :author, class_name: "User"
 
+  scope :test_categories, -> {joins(:category)}
+  scope :easy, -> {where(level: 0..1)}
+  scope :medium, -> {where(level: 2..4)}
+  scope :hard, ->{where("level>5")}
+
+  validates :level, numericality: { only_integer: true }
+  validates :title, presence: true
```

Comment:
`validates :title, presence: true, uniqueness: { scope: :level, message: "must be uniq in level"}`